### PR TITLE
Upgrade to the Lab 3 extension system

### PR DESCRIPTION
**Pull request by jtpio**
Using:

```bash
python -m jupyterlab.upgrade_extension .
```

### TODO

- [x] Upgrade to the latest beta using the script
- [x] Fix TypeScript compilation errors
- [x] Pin CI to `jupyterlab_server==2.0.0rc0` for now

### CODE REVIEW

**Comment by jtpio** [comment]
cc @hbcarlos 

Looks like CI is passing now.

Did a couple of fixes to it passes the usual ESLint and prettier checks.

Dev instructions have been updated, if you want to try the new extension system locally in a fresh environment: https://github.com/hbcarlos/voila-editor/blob/lab-3/README.md#development-install

The extension should be installable and usable, although there might be a couple of styling issues that can be fixed later: 

image

File: README.md [hunk 1]
```diff
@@ -1,69 +1,64 @@
...
 
 ```bash
 # Clone the repo to your local environment
-git clone https://github.com/...
-# Move to jupyterlab-ros directory
-cd jupyterlab-ros
+# Change directory to the voila-editor directory
+
+# create a new environment
+conda create -n voila-editor -c conda-forge python nodejs -y
```
Comment (jtpio) [inline README.md:32]: We can keep `mamba` here too

Comment (hbcarlos) [reply README.md:32]: Yep, we should use mamba.
Thanks!! I'll continue working on editor this afternoon.

Comment (jtpio) [reply README.md:32]: Nice. We can update the dev instructions in another PR (tracked in https://github.com/hbcarlos/voila-editor/issues/4)

**Pull request merged**